addons/stock/models/stock_rule.py

```diff
@@ -377,7 +377,8 @@ def _get_stock_move_values(self, product_id, product_qty, product_uom, location_
             move_values['location_dest_id'] = self.location_dest_id.id
         for field in self._get_custom_move_fields():
             if field in values:
-                move_values[field] = values.get(field)
+                value = values[field]
+                move_values[field] = value.id if isinstance(value, models.BaseModel) else value
```

Comment:
@amoyaux not sure that's the correct fix to allow `test_check_update_qty_mto_chain` to run in post_install: if sale_stock is installed, then `partner_id` is added as one of the custom_move_fields, but test_check_update_qty_mto_chain passes a recordset as a value so it fails when the recordset reaches the SQL layer.

However there are a few odd bits e.g. at the start of the function we pull the `partner` value
https://github.com/odoo-dev/odoo/blob/b7d9f161c82c40efbb75ea888a1ebccf9d5f022f/addons/stock/models/stock_rule.py#L332
and this is then set a the `partner_id`
https://github.com/odoo-dev/odoo/blob/b7d9f161c82c40efbb75ea888a1ebccf9d5f022f/addons/stock/models/stock_rule.py#L357
which this is going to override. So there are two different `value` keys being used for the same field...